internal/provider/graphql_helper.go

```diff
@@ -0,0 +1,30 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+// Helper method for modifying client requests appropriately for sending a GraphQL call instead of a REST call.
+func SendGraphQLRequest(ctx context.Context, client *gitlab.Client, graphQLCall GraphQLQuery, objectToParseForResponse interface{}) (interface{}, error) {
+
+	request, err := client.NewRequest("POST", "", graphQLCall, nil)
+	//Overwrite the path of the existing request, as otherwise the client appends /api/v4 instead.
+	request.URL.Path = "/api/graphql"
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = client.Do(request, objectToParseForResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	return objectToParseForResponse, nil
+}
```

Comment:
I did a small refactoring. WDYT?

```suggestion
func SendGraphQLRequest(ctx context.Context, client *gitlab.Client, query GraphQLQuery, response interface{}) (interface{}, error) {
	request, err := client.NewRequest("POST", "", query, nil)
	if err != nil {
		return nil, err
	}
	// Overwrite the path of the existing request, as otherwise the go-gitlab client appends /api/v4 instead.
	request.URL.Path = "/api/graphql"

	if _, err = client.Do(request, response); err != nil {
		return nil, err
	}

	return response, nil
}
```